src/test/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaWriter.java

```diff
@@ -0,0 +1,154 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.meta;
+
+import java.io.File;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+public final class XmlMetaWriter {
+    private static final Pattern FILEPATH_CONVERSION = Pattern.compile("\\.");
+
+    public void write(ModuleDetails moduleDetails) throws TransformerException,
+            ParserConfigurationException {
+        final DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        final DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+        final Document doc = dBuilder.newDocument();
+
+        final Element rootElement = doc.createElement("checkstyle-metadata");
+        final Element rootChild = doc.createElement("module");
+        rootElement.appendChild(rootChild);
+
+        doc.appendChild(rootElement);
+
+        Element checkModule = null;
+        final ModuleType moduleType = moduleDetails.getModuleType();
+        if (moduleType == ModuleType.CHECK) {
+            checkModule = doc.createElement("check");
+        }
+        else if (moduleType == ModuleType.FILTER) {
+            checkModule = doc.createElement("filter");
+        }
+        else if (moduleType == ModuleType.FILEFILTER) {
+            checkModule = doc.createElement("file-filter");
+        }
```

Comment:
instead of repeating if blocks, you can add field in enum and make it like `doc.createElement(moduleType.getElementName());`

Reply:
Done